### ISSUE TITLE
[iOS] CA1416 false positive

### ISSUE THREAD

**Issue by tipa** [dotnet/roslyn-analyzers#7578]
### Analyzer

**Diagnostic ID**: [CA1416](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416): `This call site is reachable on: 'iOS' 13.0 and later, 'maccatalyst' 13.0 and later, 'macOS/OSX' 15.0 and later. 'NSTextHighlightStyle.Default' is only supported on: 'ios' 18.0 and later, 'maccatalyst' 18.0 and later, 'tvos' 18.0 and later`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 9.0.200](https://dotnet.microsoft.com/download/dotnet/9.0)

### Describe the bug

CA1416 is incorrectly reported

### Steps To Reproduce

Solution with iOS, macOS and shared project.
This shared code is in the shared project:

```
#if IOS
using StringAttributeKey = UIKit.UIStringAttributeKey;
#else
using StringAttributeKey = AppKit.NSStringAttributeKey;
#endif

...

var attrs = new NSMutableDictionary();
if (OperatingSystem.IsIOSVersionAtLeast(18) || OperatingSystem.IsMacOSVersionAtLeast(15))
{
    attrs[StringAttributeKey.TextHighlightStyle] = NSObject.FromObject(NSTextHighlightStyle.Default.GetConstant());
    attrs[StringAttributeKey.TextHighlightColorScheme] = NSObject.FromObject(NSTextHighlightColorScheme.Orange.GetConstant());
}
```

Example project: [test.zip](https://github.com/user-attachments/files/18824790/test.zip)

### Expected behavior
No warning

### Actual behavior
warning CA1416 is incorrectly reported

## Additional context

- Problem only happens when building iOS project, but not when building macOS project

- Also checking for `OperatingSystem.IsIOS()` resolves the problem
`if (OperatingSystem.IsIOS() && OperatingSystem.IsIOSVersionAtLeast(18) || OperatingSystem.IsMacOSVersionAtLeast(15))`

- Possible duplicate of https://github.com/dotnet/roslyn-analyzers/issues/7530
